tests/RobotFramework/libraries/ThinEdgeIO/ThinEdgeIO.py

```diff
@@ -962,6 +962,67 @@ def delete_smartrest_one_template(self, template_id: str):
                 ex,
             )
 
+    @keyword("List Entities")
+    def list_entities(
+            self, name: str = None 
+    ) -> List[Dict[str, Any]]:
+        """
+        Get entity list from the device using the entity store query REST API
+
+        Args:
+            name (str, optional): Device name to fetch the entity list from
+
+        Returns:
+            List[Dict[str, Any]]: List of entities
+
+        *Example:*
+        | ${entities}= | List Entities |
+        | ${entities}= | List Entities | name=${PARENT_SN} |
+        """
+        device = self.current
+        if name:
+            if name in self.devices:
+                device = self.devices.get(name)
+
+        if not device:
+            log.info("Device has not been setup, so no logs to collect")
+            return
+
+        output = device.execute_command("curl -f http://localhost:8000/tedge/entity-store/v1/entities")
+        entities = json.loads(output.stdout)
+        return entities
+
+
+    @keyword("Assert Contains Entity")
+    def assert_contains_entity(
+            self, entity: Union[str, Dict[str, Any]], entities: List[Dict[str, Any]] = None, **kwargs
+    ) -> List[Dict[str, Any]]:
+        """Assert if the entity store contains the given entity
+
+        Args:
+            entity (str or Dict[str, Any]]): Entity to look for
+            entities (List[Dict[str, Any]], optional): List of entities to search in. Defaults to None.
+
+        Returns:
+            List[Dict[str, Any]]: List of entities matching the given entity definition
+
+        *Example:*
+        | ${entities}= | Assert Contains Entity | entity=${entity_json} |
+        | ${entities}= | Assert Contains Entity | entity=${entity_json} | entities=${entity_list_json} |
+        """
+        if not entities:
+            entities = self.list_entities()
+
+        if isinstance(entity, str):
+            entity = json.loads(entity)
+
+        matches = entities
+        if entity:
+            matches = [entity for entity in entities if entity == entity]
```

Comment:
Since renaming the function's `item` argument, it makes the list comprehension very difficult to read due to the variable shadowing happening (or I wouldn't be surprised if it does not work at all!)

Why not just keep the `item` name?

```suggestion
            matches = [entity for entity in entities if entity == item]
```

Reply:
Resolved by 55975e0